pkg/sql/information_schema.go

```diff
@@ -2470,30 +2472,54 @@ func forEachTableDescWithTableLookupInternalFromDescriptors(
 	return nil
 }
 
+type roleOptions map[string]string
+
+func (r roleOptions) noLogin() bool {
+	_, noLogin := r["NOLOGIN"]
+	return noLogin
+}
+
+func (r roleOptions) validUntil(p *planner) (*time.Time, error) {
+	rolValidUntilStr, ok := r["VALID UNTIL"]
+	var rolValidUntil *time.Time = nil
+	if ok {
+		validUntil, _, err := pgdate.ParseTimestamp(
+			p.EvalContext().GetRelativeParseTime(),
+			p.EvalContext().GetDateStyle(),
```

Comment:
we should parse this always as `ISO,MDY`. you can use `pgdate.DefaultDateStyle()`

Reply:
Take in count the following:

- We are parsing from text to timestamp, and this will run a query which will use json_object_agg that will marshal timestamp into text depending on the DateStyle format. That is the reason to unmarshal values using the same session DateStyle format.

Reply:
When do we parse text to timestamp in this? Iiuc, it is always inserted as ISO in (ymd dmy and mdy doesn't matter here) so we should translate it as ISO,mdy out.